File: backend/src/main/java/com/fee/app/schoolfeeapp/auth/service/UserManagementService.java
package com.fee.app.schoolfeeapp.auth.service;

import com.fee.app.schoolfeeapp.auth.dto.request.*;
import com.fee.app.schoolfeeapp.auth.dto.response.*;
import com.fee.app.schoolfeeapp.common.dto.PageResponse;
import org.springframework.data.domain.Pageable;
import reactor.core.publisher.Mono;

import java.util.Map;


public interface UserManagementService {

    Mono<CreateParentResponse> createParent(CreateParentRequest request);

    Mono<CreateStaffResponse> createStaff(CreateStaffRequest request);

    Mono<PageResponse<UserSummaryResponse>> listUsers(String userType, String status, String search,
                                                      Pageable pageable, String requestId);

    // Parent self-onboarding
    Mono<CheckAccountResponse> checkAccount(CheckAccountRequest request);
    Mono<Void> sendOtp(SendOtpRequest request);
    Mono<Map<String, String>> verifyOtpAndCreateAccount(VerifyOtpRequest request);
    Mono<Map<String, String>> setPassword(SetPasswordRequest request);
}

File: frontend/src/services/userManagementService.ts
import api from '@/lib/api';
import type { 
  CreateStaffRequest, 
  CreateParentRequest,
  PageResponse,
  ApiResponse,
  UserSummaryResponse
} from '@/types/user-management';

export const userManagementService = {
  getUsers: async (params: { page: number; size: number; userType?: string; status?: string; search?: string; sortBy?: string }): Promise<PageResponse<UserSummaryResponse>> => {
    const { data } = await api.get<ApiResponse<PageResponse<UserSummaryResponse>>>('/api/v1/auth/users', { params });
    return data.data;
  },

  createStaff: async (requestData: CreateStaffRequest): Promise<any> => {
    const { data } = await api.post<ApiResponse<any>>('/api/v1/auth/staff', requestData);
    return data.data;
  },

  createParent: async (requestData: CreateParentRequest): Promise<any> => {
    const { data } = await api.post<ApiResponse<any>>('/api/v1/auth/parents', requestData);
    return data.data;
  }
};
